include/verilated.cpp

```diff
@@ -3098,3 +3098,15 @@ void VerilatedAssertOneThread::fatal_different() VL_MT_SAFE {
 #endif
 
 //===========================================================================
+// VlDeleter:: Methods
+
+void VlDeleter::deleteAll() {
+    // No lock needed, as it all happens on one thread
```

Comment:
```suggestion
    // No lock needed, as it all happens on one thread
    VL_DEBUG_IF(static VerilatedAssertOneThread;);
```

Alternatively though true the cost of getting a lock with only a single consumer is extremely low, so might do it anyways.